### PR TITLE
Home_work Subway System problem

### PR DESCRIPTION
**Pull request by badraih**

### CODE REVIEW

**Review by mfalthaw — COMMENTED** [review]
Hi @badraih, I just want to start by saying, wow, this is incredible. The amount of work and effort you must have put into this is amazing! You should be proud of the work and commitment. Here are some of my comments:
**Props**
- I ran your code, and it does work perfectly with a couple of tests, bravo!
- Awesome use of comments! I'd love to see all code out there commented the same way
- Great job following namingConvention
- I love the fact that you used an object to hold all of the line stops


**Areas of improvement**
- When code gets bigger, we always try to avoid putting everything in one function, this will make our code harder to read and tougher to debug

- As much as you possibly can, try to keep your code [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

- One of the advantages of using object to hold our data, is that we can use its keys to easily get the values we want. So, instead of having to do 

```js
if (line1 === "N") {
        for (var i = 0; i < subwaySystem.firstLine.length; i++) {
        	....
        }//end loop
      }// end if line1 equal "N"
```

We can use 

```js
var startLine = subwaySystem.line1;
var endLine = subwaySystem.line2;
var startStopIndex = startLine.indexOf(startPoint);
var endStopIndex = endLine.indexOf(endPoint);

        for (var i = startStopIndex; i < endStopIndex; i++) {
        	....
        }//end loop
      }// end if line1 equal "N"
```

This will reduce our code by **85%** because it will work for any valid line and any valid stop. It will also eliminate the need to check which line is `line1` and which line is `line2`


- To check if two lines are the same, instead of doing 
```js
if (line1 === "N" && line2 === "N" || line1 === "6" && line2 === "6" || line1 === "L" && line2 === "L")
```

You can do 
```js
if (line1 === line2)
```


In conclusion, very good work! Please remind me to give you a sticker of your choice for a job well-done!
